src/dso_api/dynamic_api/filterset.py

```diff
@@ -37,29 +49,61 @@ def get_filters(cls):
     # See https://django-filter.readthedocs.io/en/master/guide/usage.html on how filters are used.
     # Determine which fields are included:
     # Excluding geometry fields for now, as the default filter only performs exact matches.
     # This isn't useful for polygon fields, and excluding it avoids support issues later.
-    fields = {f.attname: _get_field_lookups(f) for f in model._meta.get_fields()}
+    fields = {f.attname: _get_field_lookups(f.__class__) for f in model._meta.get_fields()}
+
+    filters = {}
+    camel_case_fields = list(map(snake_to_camel_case, fields))
+    for field_name in serializer_class.Meta.fields:
+        if field_name not in ["_links", "schema"] + camel_case_fields:
+            try:
+                field_schema = model._table_schema['schema']['properties'][field_name]
+            except KeyError:
+                pass
+            else:
+                if field_is_nested_table(field_schema):
+                    for inner_name, inner_field in field_schema['items']['properties'].items():
+                        model_field = JSON_TYPE_TO_DJANGO[inner_field['type']][0]
+                        filter_class = DSOFilterSet.FILTER_DEFAULTS.get(model_field)
+                        if filter_class is not None:
+                            filter_class = filter_class["filter_class"]
+                            filter_name = "__".join([field_name, format_field_name(inner_name)])
+                            field_lookups = _get_field_lookups(model_field)
+                            for lookup_expr in field_lookups:
+                                subfilter_name = filter_name
+                                if lookup_expr != 'exact':
+                                    subfilter_name = f"{filter_name}__{lookup_expr}"
+                                filters[subfilter_name] = filter_class(
+                                    field_name="__".join([field_name, slugify(inner_name, sign='_')]),
+                                    lookup_expr=lookup_expr if lookup_expr != 'exact' else 'contains',
+                                    label=DSOFilterSet.FILTER_HELP_TEXT.get(filter_class, lookup_expr)
+                                )
+                                fields[subfilter_name] = field_lookups
 
-    # Generate the class
     meta_attrs = {
         "model": model,
         "fields": fields,
+        "serializer_class": serializer_class,
     }
     meta = type("Meta", (), meta_attrs)
-    return type(f"{model.__name__}FilterSet", (DynamicFilterSet,), {"Meta": meta})
+    return type(
+        f"{model.__name__}FilterSet", (DynamicFilterSet,), {"Meta": meta, **filters}
+    )
 
 
-def _get_field_lookups(field: models.Field) -> list:
+def _get_field_lookups(field_class) -> list:
     """Find the possible lookups for a given field type."""
-    return DEFAULT_LOOKUPS_BY_TYPE.get(field.__class__, ["exact"])
+    return DEFAULT_LOOKUPS_BY_TYPE.get(field_class, ["exact"])
```

Comment:
ik had `field_class` hier staan, en op suggestie van de diff veranderd in `field.__class__`. Mogelijk is dat bij relaties straks wel nodig om het field zelf door te kunnen lezen. Ik zou dit dus niet terugveranderen.

Reply:
Weet ik, helaas in die grote blok text boven hebben we geen `field` maar allen de `class` die  hoort erbij.